### PR TITLE
Adding new interface for organization deletion

### PR DESCRIPTION
**Pull request by HiranyaKavishani**
This change will add a new interface to delete org collection

### CODE REVIEW

File: components/apimgt/org.wso2.carbon.apimgt.persistence/src/main/java/org/wso2/carbon/apimgt/persistence/APIPersistence.java [hunk 1]
```diff
@@ -144,6 +144,14 @@ void deleteAPIRevision(Organization org, String apiUUID, String revisionUUID, in
      */
     void deleteAPI(Organization org, String apiId) throws APIPersistenceException;
 
+    /**
+     * Delete APIs
+     *
+     * @param org   Organization the API is owned by
+     * @throws APIPersistenceException
+     */
+    void deleteAPIs(Organization org) throws APIPersistenceException;
```
Comment (chamilaadhi) [inline components/apimgt/org.wso2.carbon.apimgt.persistence/src/main/java/org/wso2/carbon/apimgt/persistence/APIPersistence.java:153]: if you add deleteAPIs , then you need to pass list of apis. if you want to delete all the apis  then i think name should be deleteAllAPIs()

File: components/apimgt/org.wso2.carbon.apimgt.persistence/src/main/java/org/wso2/carbon/apimgt/persistence/RegistryPersistenceImpl.java [hunk 2]
```diff
@@ -875,6 +875,11 @@ public void deleteAPI(Organization org, String apiId) throws APIPersistenceExcep
         }
     }
 
+    @Override
+    public void deleteAPIs(Organization org) throws APIPersistenceException {
+
```
Comment (chamilaadhi) [inline components/apimgt/org.wso2.carbon.apimgt.persistence/src/main/java/org/wso2/carbon/apimgt/persistence/RegistryPersistenceImpl.java:880]: add a comment for keeping it empty

File: components/apimgt/org.wso2.carbon.apimgt.persistence/src/main/java/org/wso2/carbon/apimgt/persistence/APIPersistence.java (hunk 1 above)
Comment (HiranyaKavishani) [reply components/apimgt/org.wso2.carbon.apimgt.persistence/src/main/java/org/wso2/carbon/apimgt/persistence/APIPersistence.java:153]: No. we can just delete whole collection calling following way. We don't need the API list
MongoCollection<MongoDBPublisherAPI> collection = MongoDBConnectionUtil.getPublisherCollection(org.getName());
collection.drop();

File: components/apimgt/org.wso2.carbon.apimgt.persistence/src/main/java/org/wso2/carbon/apimgt/persistence/APIPersistence.java [hunk 3]
```diff
@@ -144,6 +144,14 @@ void deleteAPIRevision(Organization org, String apiUUID, String revisionUUID, in
      */
     void deleteAPI(Organization org, String apiId) throws APIPersistenceException;
 
+    /**
+     * Delete all APIs
+     *
+     * @param org   Organization the API is owned by
+     * @throws APIPersistenceException
+     */
+    void deleteAllAPIs(Organization org) throws APIPersistenceException;
```
Comment (malinthaprasan) [inline components/apimgt/org.wso2.carbon.apimgt.persistence/src/main/java/org/wso2/carbon/apimgt/persistence/APIPersistence.java:153]: This is not only about APIs right.. Even thought we consider about APIs only now, objective of this method is to delete anything that belongs to the organization? 
```suggestion
    void purge(Organization org) throws APIPersistenceException;
```

File: components/apimgt/org.wso2.carbon.apimgt.persistence/src/main/java/org/wso2/carbon/apimgt/persistence/APIPersistence.java (hunk 1 above)
Comment (malinthaprasan) [reply components/apimgt/org.wso2.carbon.apimgt.persistence/src/main/java/org/wso2/carbon/apimgt/persistence/APIPersistence.java]: This is renamed right @HiranyaKavishani

File: components/apimgt/org.wso2.carbon.apimgt.persistence/src/main/java/org/wso2/carbon/apimgt/persistence/APIPersistence.java (hunk 3 above)
Comment (malinthaprasan) [reply components/apimgt/org.wso2.carbon.apimgt.persistence/src/main/java/org/wso2/carbon/apimgt/persistence/APIPersistence.java:153]: Please update the discussion points we had @HiranyaKavishani

**Review by malinthaprasan — APPROVED** [review]

File: components/apimgt/org.wso2.carbon.apimgt.persistence/src/main/java/org/wso2/carbon/apimgt/persistence/APIPersistence.java (hunk 3 above)
Comment (HiranyaKavishani) [reply components/apimgt/org.wso2.carbon.apimgt.persistence/src/main/java/org/wso2/carbon/apimgt/persistence/APIPersistence.java:153]: Yes. But as per the organization deletion use case, the deletion of the MongoDB collection is done under the APIPurge functionality.So a better name should be similar to `deleteAllAPI`

**Review by chamilaadhi — APPROVED** [review]

**Pull request merged**